### PR TITLE
Implemented obstacle avoidance with Smartcar

### PR DESCRIPTION
**Pull request by Soarnir**

### CODE REVIEW

File: arduino/smartcar/smartcar.ino [hunk 1]
```diff
@@ -1,9 +1,45 @@
...
+const unsigned int offset = 0;
+
+//Car creation
+ArduinoRuntime arduinoRuntime;
+BrushedMotor leftMotor{arduinoRuntime, smartcarlib::pins::v2::leftMotorPins};
+BrushedMotor rightMotor{arduinoRuntime, smartcarlib::pins::v2::rightMotorPins};
+DifferentialControl control{leftMotor, rightMotor};
+DirectionalOdometer odometer(arduinoRuntime, smartcarlib::pins::v2::leftOdometerPins, []() {odometer.update();}, PULSES_PER_METER);
+GY50 gyroscope(arduinoRuntime, offset);
+SR04 ultrasonic(arduinoRuntime, TRIGGER_PIN, ECHO_PIN, MAX_DISTANCE);
```
Comment (platisd) [inline arduino/smartcar/smartcar.ino:17]: The maximum distance the sensor can detect is `400`

**Review by platisd — APPROVED** [review]

File: arduino/smartcar/smartcar.ino (hunk 1 above)
Comment (Soarnir) [reply arduino/smartcar/smartcar.ino:17]: Noted, thanks. Seems like something which would be useful to have specified in the [API docs](https://platisd.github.io/smartcar_shield/class_s_r04.html)?

Comment (platisd) [reply arduino/smartcar/smartcar.ino:17]: It's already in the [SR04 instruction manuals](https://lastminuteengineers.com/arduino-sr04-ultrasonic-sensor-tutorial/) that are widely available and on [smce-gd documentation](https://github.com/ItJustWorksTM/smce-gd/wiki/Vehicle-Capabilities#ultrasonic-distance-sensor).

Comment (Soarnir) [reply arduino/smartcar/smartcar.ino:17]: Good point, thanks for the links.

**Pull request merged**